tendrl/alerting/api/manager.py

```diff
@@ -0,0 +1,114 @@
+from flask import Flask
+from flask import request
+from flask import Response
+import logging
+from multiprocessing import Event
+from multiprocessing import Process
+from tendrl.alerting.etcd.exceptions import EtcdError
+from tendrl.alerting.etcd.manager import EtcdManager
+from tendrl.alerting.exceptions import AlertingError
+from tendrl.alerting.exceptions import InvalidRequest
+from tendrl.alerting.notification.exceptions import InvalidHandlerConfig
+from tendrl.alerting.notification.exceptions import NotificationPluginError
+from tendrl.alerting.notification.manager import NotificationPluginManager
+from tendrl.common.config import TendrlConfig
+
+
+config = TendrlConfig()
+LOG = logging.getLogger(__name__)
+app = Flask(__name__)
+
+
+@app.route("/alerting/notification_medium/supported")
+def get_handlers():
+    try:
+        return Response(
+            NotificationPluginManager().get_handlers(),
+            mimetype='application/json'
+        )
+    except (
+        AlertingError,
+        NotificationPluginError
+    ) as ex:
+        raise ex
+
+
+@app.route("/alerting/notification_medium/<name>/help")
+def get_config_help(name):
+    try:
+        return Response(
+            NotificationPluginManager().get_config_help(name),
+            mimetype='application/json'
+        )
+    except (
+        InvalidRequest,
+        NotificationPluginError
+    ) as ex:
+        raise ex
+
+
+@app.route("/alerting/alerts")
+def get_alerts(filter_criteria=None):
```

Comment:
this is basically used for getting alerts by node_id, severity, etc.. (any field of alert). This goes all the way to https://github.com/Tendrl/alerting/pull/20/files#diff-b5f33c2b8d8f1b2d05e255cb1314a7d2 line no. 60 in which the actual filtering happens